app/controllers/media_controller.rb

```diff
@@ -19,18 +19,29 @@ def stream
     authorize! :read, @media
     respond_to do |format|
       format.m3u8 do
-        redirect_to @media.to_playlist_url
+        redirect_to "#{@media.to_playlist_url}?token_string=#{media_token}"
```

Comment:
yeah, i think jessie and i both felt that it looked a bit inelegant, but not terrible, and seems like everyone can live with it (and no one seems to have an obvious better solution).